### PR TITLE
Update to solidity 0.5

### PR DESCRIPTION
**Pull request by spalladino**
This PR builds on top of @paulinablaszk great work in https://github.com/OpenZeppelin/ethernaut/pull/123, updating the necessary dependencies so `deploy:contracts` works.

### CODE REVIEW

**Comment by abcoathup** [comment]
Hi @spalladino,

Unfortunately not working on Windows Subsystem for Linux.
Netlify deploy preview failed too.

Hangs on writing deploy data when deploying contracts locally.  This worked previously. 
```
$ npm run deploy:contracts 

...

Writing updated deploy data: ./gamedata/deploy.local.json
```
Fails on starting dapp.  Different error than previously, though assume still need some transpiling.
```
$ npm start 

...

Failed to compile.

Error in ./src/reducers/gamedataReducer.js
Syntax error: Unexpected token (14:8)

  12 |     case actions.LOAD_GAME_DATA:
  13 |       return {
> 14 |         ...state,
     |         ^
  15 |         levels: action.levels,
  16 |         ethernautAddress: action.ethernautAddress
  17 |       }

 @ ./src/reducers/index.js 15:23-51

Error in ./src/reducers/networkReducer.js
Syntax error: Unexpected token (14:15)

  12 |
  13 |     case actions.CONNECT_WEB3:
> 14 |       return { ...state, web3: action.web3 }
     |                ^
  15 |
  16 |     case actions.SET_GAS_PRICE:
  17 |       return { ...state, gasPrice: action.gasPrice }
```

**Comment by spalladino** [comment]
Just updated it! Can you re-test please?

File: package.json [hunk 1]
```diff
@@ -43,15 +48,15 @@
...
     "path-exists": "2.1.0",
     "postcss-loader": "1.0.0",
     "promise": "7.1.1",
     "react-dev-utils": "^0.4.2",
     "recursive-readdir": "2.1.0",
     "strip-ansi": "3.0.1",
     "style-loader": "0.13.1",
-    "truffle-contract": "^3.0.0",
-    "truffle-solidity-loader": "0.0.8",
+    "truffle": "^5.0.14",
```
Comment (gnidan) [inline package.json]: Not 5.1?

**Comment by abcoathup** [comment]
Hi @spalladino,

Still hangs on writing deploy data when deploying contracts locally.  In `ganache-cli` there are repeated `eth_getBlockByNumber`.
```
$ npm run deploy:contracts 

...

Writing updated deploy data: ./gamedata/deploy.local.json
```

Can run `npm start`

**Comment by spalladino** [comment]
Yep, I think the hung process is due to either the truffle or web3 update, where it some connections are left opened and the process never finishes. I'd ignore them tbh.

I'll merge this and make an stg deploy to better test it out!

File: package.json (hunk 1 above)
Comment (spalladino) [reply package.json]: Easy mate, we are already upgrading a major version! One step at a time :-P

**Pull request merged**